### PR TITLE
Add migration guide and update analytics API reference for AEPAnalytics

### PR DESCRIPTION
**Pull request by cacheung**
Add migration guide for Analytics
Update analytics API reference with AEPAnalytics.

### CODE REVIEW

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 1]
```diff
@@ -1009,7 +1150,36 @@ Analytics.sendQueuedHits();
 ```
 {% endtab %}
 
-{% tab title="iOS" %}
+{% tab title="iOS (AEP 3.x)" %}
+
+### sendQueuedHits
+
+Regardless of how many hits are currently queued, this method forces the library to send all hits in the offline queue.
```
Comment (calebk1) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:1157]: This method forces the library to send all hits in the offline queue, regardless of how many hits are currently queued.

File: using-mobile-extensions/adobe-analytics/migration.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
+# Migrating to AEPAnalytics reference
+
+This document is a reference comparison of AEPAnalytics (3.x) APIs against their equivalent ACPAnalytics(2.x) APIs.
```
Comment (calebk1) [inline using-mobile-extensions/adobe-analytics/migration.md:3]: Missing a space between ACPAnalytics and (2.x).

**Review by calebk1 — COMMENTED** [review]
A few comments.

For all the areas with `**Warning:**`, the first comment (about using the 
```
{% hint style="warning" %}
{% endhint %
``` 
sytnax) apply.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 3]
```diff
@@ -97,7 +113,35 @@ Analytics.clearQueue();
 ```
 {% endtab %}
 
-{% tab title="iOS" %}
+{% tab title="iOS (AEP 3.x)" %}
+
+### clearQueue
+
+**Warning:** Use caution when manually clearing the queue. This process cannot be reversed.
```
[the remark this answers is not in the document]
Comment (cacheung) [reply using-mobile-extensions/adobe-analytics/analytics-api-reference.md:120]: This is inside of the code tab.  Several places have been using **Warning:**  format.  
e.g: https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics/analytics-api-reference#clearqueue
(For Android and IOS)

Let me know if you think is better to use the syntax, I will change for them.

[the remark this answers is not in the document]
Comment (calebk1) [reply using-mobile-extensions/adobe-analytics/analytics-api-reference.md:120]: Yeah, I saw that - but I'm curious why we don't have the warning hint inside the tab.

Does that not work? I just genuinely don't know, that's all :P 

(If it does work, I'd prefer to use the warning hint. If it doesn't work, then just use the \**Warning**.)

[the remark this answers is not in the document]
Comment (cacheung) [reply using-mobile-extensions/adobe-analytics/analytics-api-reference.md]: I updated them with hint format.  I can't see the output in the PR view.  I will verify it once it is merged.

**Review by calebk1 — APPROVED** [review]
LGTM!

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 4]
```diff
@@ -434,7 +525,39 @@ class StringCallback : Java.Lang.Object, IAdobeCallback
...
+
+```objectivec
+[AEPMobileAnalytics getQueueSize:^(NSInteger queueSize, NSError * _Nullable error) {
+    // handle queue size
+ }];
+```
+
+{% endtab %}
+
+{% tab title="iOS (ACP 2.x)" %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:559]: Is this section duplicated? I see getQueueSize above.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 5]
```diff
@@ -506,7 +629,39 @@ Analytics.getTrackingIdentifier(new AdobeCallback<String>() {
...
+static func getTrackingIdentifier(completion: @escaping (String?, Error?) -> Void)
+```
+
+#### Example
+
+**Swift**
+
+```swift
+Analytics.getTrackingIdentifier { (trackingId, error) in
+   // check the trackingIdentifier value
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:650]: // Handle the error (if non-nil) or use the trackingId value.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 6]
```diff
@@ -506,7 +629,39 @@ Analytics.getTrackingIdentifier(new AdobeCallback<String>() {
...
+Analytics.getTrackingIdentifier { (trackingId, error) in
+   // check the trackingIdentifier value  
+}
+```
+
+**Objective-C**
+
+```objectivec
+AEPMobileAnalytics getTrackingIdentifier:^(NSString * _Nullable trackingIdentifier, NSError * _Nullable error) {
+   // check the trackingIdentifier value
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:658]: // Handle the error (if non-nil) or use the trackingIdentifier value.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 7]
```diff
@@ -705,7 +860,41 @@ If you have an [Experience Cloud ID](https://app.gitbook.com/@aep-sdks/s/docs/us
...
+
+**Objective-C**
+
+```objectivec
+AEPMobileAnalytics getTrackingIdentifier:^(NSString * _Nullable trackingIdentifier, NSError * _Nullable error) {
+   // check the trackingIdentifier value  
+}];
+```
+
+{% endtab %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:894]: Same comment here regarding the duplicated section.

If we only have one API in AEP, one thing we can do is to document it as part of the getTrackingIdentifier section and just provide a link to it in the getTrackingIdentifierWithCompletionHandler section.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 8]
```diff
@@ -953,7 +1174,41 @@ Before using this API, see [Identify unique visitors](https://docs.adobe.com/con
...
+
+**Objective-C**
+
+```objectivec
+[AEPMobileAnalytics getVisitorIdentifier:^(NSString * _Nullable visitorIdentifier, NSError * _Nullable error) {
+    // check the visitorIdentifier value or handle error
+}];
+```
+
+{% endtab %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:1208]: Same here, let's not duplicate this code and reference the getVisitorIdentifier section instead.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 9]
```diff
@@ -82,7 +99,9 @@ Force delete, without sending to Analytics, all hits being stored or batched on
 {% tab title="Android" %}
 ### clearQueue
 
-**Warning:** Use caution when manually clearing the queue. This process cannot be reversed.
+{% hint style="warning" %}
+Use caution when manually clearing the queue. This process cannot be reversed.
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:103]: I think this is meant to be: `... This operation cannot be reverted.`
Let's update it everywhere.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 10]
```diff
@@ -253,7 +314,37 @@ Analytics.getQueueSize(new AdobeCallback<Long>() {
...
+static func getQueueSize(completion: @escaping (Int, Error?) -> Void)
+```
+
+#### Example
+
+**Swift**
+
+```swift
+Analytics.getQueueSize { (queueSize, error) in
+    // handle queue size
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:333]: // Handle error (if non-nil) or use queueSize.
- same comment below

Comment (cacheung) [reply using-mobile-extensions/adobe-analytics/analytics-api-reference.md]: Thanks, Updated.  @emdobrin.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 11]
```diff
@@ -208,16 +208,14 @@ Future<void> clearQueue();
...
 {% hint style="warning" %}
-Use caution when manually clearing the queue. This process cannot be reversed.
+Use caution when manually clearing the queue. This operation cannot be reverted.
 {% endhint %}
 
 **Example**
 
 ```dart
-FlutterACPAnalytics.clearQueue();
+FlutterACPAnalytics.clearQueue();{% endtab %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:217]: this needs to be closed before L219

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 12]
```diff
@@ -705,7 +813,41 @@ If you have an [Experience Cloud ID](https://app.gitbook.com/@aep-sdks/s/docs/us
...
+Analytics.getTrackingIdentifier { (trackingId, error) in
+   // Handle the error (if non-nil) or use the trackingId value. 
+}
+```
+
+**Objective-C**
+
+```objectivec
+AEPMobileAnalytics getTrackingIdentifier:^(NSString * _Nullable trackingIdentifier, NSError * _Nullable error) {
+   // Handle the error (if non-nil) or use the trackingId value.
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:843]: .. or use the trackingIdentifier value

Please note that this comment talks about the trackingIdentifier parameter. Make sure all comments below follow this rule.

File: using-mobile-extensions/adobe-analytics/analytics-api-reference.md [hunk 13]
```diff
@@ -953,7 +1104,39 @@ Before using this API, see [Identify unique visitors](https://docs.adobe.com/con
...
+Analytics.getVisitorIdentifier { (visitorId, error) in
+   // check the visitorIdentifier value or handle error
+}
+```
+
+**Objective-C**
+
+```objectivec
+[AEPMobileAnalytics getVisitorIdentifier:^(NSString * _Nullable visitorIdentifier, NSError * _Nullable error) {
+    // check the visitorIdentifier value or handle error
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics/analytics-api-reference.md:1132]: let's reverse this `// Handle the error (if non-nil) or use the visitorIdentifier value`

**Review by emdobrin — COMMENTED** [review]
@cacheung thanks for the update, looks great with few minor things

File: using-mobile-extensions/adobe-analytics/migration.md [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+{% tab title="ACP 2.x (Objective-C)" %}
+
+```objective-c
++ (void) clearQueue;
+```
+
+{% endtab %}
+{% endtabs %}
+
+### getQueueSize
```
[the remark this answers is not in the document]
Comment (cacheung) [reply using-mobile-extensions/adobe-analytics/migration.md:76]: Sorted.  Thanks.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**